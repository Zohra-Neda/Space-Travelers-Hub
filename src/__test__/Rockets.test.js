import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Rockets from '../pages/Rockets';

describe('Rockets component', () => {
  const mockStore = configureStore([]);

  test('renders correctly', () => {
    const initialState = {
      rockets: {
        loading: false,
        value: [
          {
            id: 'rocket1',
            name: 'Rocket 1',
            description: 'This is rocket 1',
            image: 'https://example.com/rocket1.jpg',
            reserved: false,
          },
          {
            id: 'rocket2',
            name: 'Rocket 2',
            description: 'This is rocket 2',
            image: 'https://example.com/rocket2.jpg',
            reserved: true,
          },
        ],
        error: '',
      },
    };

    const store = mockStore(initialState);

    const component = renderer.create(
      <Provider store={store}>
        <Rockets />
      </Provider>,
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
