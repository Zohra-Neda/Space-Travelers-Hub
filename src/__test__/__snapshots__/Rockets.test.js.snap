// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rockets component renders correctly 1`] = `
<div
  className="rocket-container"
>
  <div
    className="rocket-card flex gap-10 my-7 w-[80%] mx-auto"
  >
    <figure
      className="img-container"
    >
      <img
        alt="rocket"
        className="rocket-image max-w-none"
        src="https://example.com/rocket1.jpg"
      />
    </figure>
    <div
      className="rocket-data flex flex-col gap-3"
    >
      <h2
        className=" font-semibold text-2xl rocket-title"
      >
        Rocket 1
      </h2>
      <p
        className="rocket-description text-justify"
      >
        This is rocket 1
         
      </p>
      <button
        className="reserve-btn px-4 py-2 w-[20%] text-white bg-blue-500 rounded-sm shadow-md hover:bg-blue-600 focus:outline-none focus:ring-2 focus:ring-blue-500 focus:ring-opacity-50"
        onClick={[Function]}
        type="button"
      >
        Reserved
      </button>
    </div>
  </div>
  <div
    className="rocket-card flex gap-10 my-7 w-[80%] mx-auto"
  >
    <figure
      className="img-container"
    >
      <img
        alt="rocket"
        className="rocket-image max-w-none"
        src="https://example.com/rocket2.jpg"
      />
    </figure>
    <div
      className="rocket-data flex flex-col gap-3"
    >
      <h2
        className=" font-semibold text-2xl rocket-title"
      >
        Rocket 2
      </h2>
      <p
        className="rocket-description text-justify"
      >
        <span
          className=" text-white bg-green-500 px-2 rounded-sm"
        >
          Reserved
        </span>
         This is rocket 2
         
      </p>
      <button
        className=" border border-2 text-gray-500 rounded-sm reserve-btn px-2 py-2 w-[20%]"
        onClick={[Function]}
        type="button"
      >
        Cancel Reservation
      </button>
    </div>
  </div>
</div>
`;
